### PR TITLE
gbox -> geobox

### PR DESCRIPTION
**Pull request by uchchwhash**
### Reason for this pull request
https://github.com/opendatacube/datacube-core/issues/1032

Will it break other people's code? Certainly.
Was it trivial? Yes.
Do I like this still? Yes.

### CODE REVIEW

File: datacube/utils/geometry/_warp.py [hunk 1]
```diff
@@ -118,8 +118,8 @@ def warp_affine(src: np.ndarray,
 
 def rio_reproject(src: np.ndarray,
                   dst: np.ndarray,
-                  s_gbox: GeoBox,
-                  d_gbox: GeoBox,
+                  s_geobox: GeoBox,
```
Comment (omad) [inline datacube/utils/geometry/_warp.py:121]: these should change to `src_geobox` and `dst_geobox`

**Review by omad — CHANGES_REQUESTED** [review]
I do like this name, but I'm hesitant about making abrupt API changes.

We don't have a policy, but I think we need a deprecation process of:
- Continue to allow the use of the old names, but throw warnings
- allow the new terms to be used.

Then in a successive release we can clean it all up properly.

Also, a better summary in the PR description would be good. eg:
> Renamed `gbox` to `geobox` everywhere, including
> - `datacube.utils.geometry.gbox` to `datacube.utils.geometry.geobox`
> - All parameter names including `gbox`.
> - ...

**Comment by robbibt** [comment]
I really, really like this change - but agree with @omad that it has the potential to break a lot of code!

**Comment by uchchwhash** [comment]
Agreed, but that's a lot of work! 🤣 
Let's just leave it here until someone finds the time to do it properly, but at least we can agree that we need this.

**Pull request closed without merging**